drivers/adc/adc_nrfx_saadc.c

```diff
@@ -358,6 +360,17 @@ static void saadc_irq_handler(void *param)
 {
 	struct device *dev = (struct device *)param;
 
+	if (nrf_saadc_event_check(NRF_SAADC_EVENT_CALIBRATEDONE)) {
+		nrf_saadc_event_clear(NRF_SAADC_EVENT_CALIBRATEDONE);
+
+		/* PAN-86 and PAN-178 require explicit STOP before START */
+		nrf_saadc_event_clear(NRF_SAADC_EVENT_STOPPED);
+		nrf_saadc_task_trigger(NRF_SAADC_TASK_STOP);
+
+		nrf_saadc_event_clear(NRF_SAADC_EVENT_END);
```

Comment:
No, but if there's a stray `EVENTS_END` that's still pending then the sample operation will be aborted in the same ISR call.  I'm not entirely convinced that CALIBRATEOFFSET doesn't generate an END in some situations, esp. on nRF52832.  I'm not going to test that, so please just confirm whether you want the clear left in place or removed.

Reply:
I think it should be removed then. And I think the events should be handled in this way:
```
if (nrf_saadc_event_check(NRF_SAADC_EVENT_END)) {
...
} else if (nrf_saadc_event_check(NRF_SAADC_EVENT_CALIBRATEDONE)) {
...
}
```
since that's how we expected them to occur (mostly the `END` one and never both at the same time).